src/appMain/life_cycle.cc

```diff
@@ -362,30 +362,24 @@ bool LifeCycle::InitMessageSystem() {
 #endif  // MQUEUE_HMIADAPTER
 
 namespace {
-pthread_t main_thread;
-void sig_handler(int sig) {
-  switch (sig) {
-    case SIGINT:
-      LOG4CXX_DEBUG(logger_, "SIGINT signal has been caught");
-      break;
-    case SIGTERM:
-      LOG4CXX_DEBUG(logger_, "SIGTERM signal has been caught");
-      break;
-    case SIGSEGV:
-      LOG4CXX_DEBUG(logger_, "SIGSEGV signal has been caught");
-      break;
-    default:
-      LOG4CXX_DEBUG(logger_, "Unexpected signal has been caught");
-      break;
-  }
-  /*
-   * Resend signal to the main thread in case it was
-   * caught by another thread
-   */
-  if (pthread_equal(pthread_self(), main_thread) == 0) {
-    LOG4CXX_DEBUG(logger_, "Resend signal to the main thread");
-    if (pthread_kill(main_thread, sig) != 0) {
-      LOG4CXX_FATAL(logger_, "Send signal to thread error");
+  void sig_handler(int sig) {
+    switch(sig) {
+      case SIGINT:
+        LOG4CXX_DEBUG(logger_, "SIGINT signal has been caught");
+        break;
+      case SIGTERM:
+        LOG4CXX_DEBUG(logger_, "SIGTERM signal has been caught");
+        break;
+      case SIGSEGV:
+        LOG4CXX_DEBUG(logger_, "SIGSEGV signal has been caught");
+        FLUSH_LOGGER();
+        // exit need to prevent endless sending SIGSEGV
+        // http://stackoverflow.com/questions/2663456/how-to-write-a-signal-handler-to-catch-sigsegv
```

Comment:
@anosach-luxoft , I think no, because it abnormal finishing of program. And in the end in any cases all memory will be free without DEINIT_LOGGER.